File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/view/DrawView.java
package com.savvasdalkitsis.gameframe.draw.view;

import com.savvasdalkitsis.gameframe.grid.model.Grid;

import rx.functions.Action1;

public interface DrawView {

    void askForFileName(Action1<String> name);

    void fileUploaded();

    void failedToUpload(Throwable e);

    void displayUploading();

    void drawingAlreadyExists(String name, Grid colorGrid, Throwable e);

    void failedToDelete(Throwable e);
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/view/DrawView.kt
package com.savvasdalkitsis.gameframe.draw.view

import com.savvasdalkitsis.gameframe.grid.model.Grid
import com.savvasdalkitsis.gameframe.infra.kotlin.TypeAction

interface DrawView {

    fun askForFileName(name: TypeAction<String>)

    fun fileUploaded()

    fun failedToUpload(e: Throwable)

    fun displayUploading()

    fun drawingAlreadyExists(name: String, colorGrid: Grid, e: Throwable)

    fun failedToDelete(e: Throwable)
}
